ui/apps/platform/src/Containers/Clusters/InitBundles/RevokeBundleModal.tsx

```diff
@@ -0,0 +1,153 @@
+                    setHasMoreClusters(false);
+                    onCloseModal(true);
+                } else {
+                    // The bundle has more impacted clusters than the list already rendered.
+                    // Therefore, user needs to confirm revoke again.
+                    setHasMoreClusters(true);
+                    setImpactedClusters(initBundleRevocationErrors[0].impactedClusters);
+                }
+            })
+            .catch((error) => {
+                setErrorMessage(getAxiosErrorMessage(error));
+            })
+            .finally(() => {
+                setIsRevokingBundle(false);
+            });
+    }
+
+    function onCancel() {
+        setErrorMessage('');
+        setHasMoreClusters(false);
+        onCloseModal(false);
+    }
+
+    // showClose={false} to prevent clicking close while isRevokingBundle.
+    return (
+        <Modal
+            title="Revoke cluster init bundle"
+            variant="small"
+            isOpen
+            showClose={false}
+            actions={[
+                <Button
+                    key="Revoke bundle"
+                    variant={impactedClusters.length === 0 ? 'primary' : 'danger'}
+                    onClick={onRevokeBundle}
+                    isDisabled={isRevokingBundle}
+                >
+                    Revoke bundle
+                </Button>,
+                <Button
+                    key="Cancel"
+                    variant="secondary"
+                    onClick={onCancel}
+                    isDisabled={isRevokingBundle}
+                >
+                    Cancel
+                </Button>,
+            ]}
+        >
+            <Flex direction={{ default: 'column' }}>
+                <DescriptionList isHorizontal>
+                    <DescriptionListGroup>
+                        <DescriptionListTerm>Bundle name</DescriptionListTerm>
+                        <DescriptionListDescription>{initBundle.name}</DescriptionListDescription>
+                    </DescriptionListGroup>
+                </DescriptionList>
+                {impactedClusters.length === 0 ? (
+                    <Alert
+                        title="No secured clusters depend on this bundle"
+                        variant="info"
+                        isInline
+                        component="p"
+                    />
+                ) : (
+                    <>
+                        {hasMoreClusters ? (
+                            <Alert
+                                title="Additional secured clusters now depend on this initBundle"
+                                variant="warning"
+                                isInline
+                                component="p"
+                            >
+                                You must confirm <strong>again</strong> to delete this bundle.
+                            </Alert>
+                        ) : (
```

Comment:
I haven't seen this pattern before.

Did we set it up this way because some clusters may have more clusters than can be displayed clearly in the modal?

Reply:
Good eyes. It is for edge case that `impactedClusters` has change between initial GET /v1/cluster-init/init-bundles request and PATCH /v1/cluster-init/init-bundles/revoke request.

Reply:
Concerning your hypothesis, I temporarily duplicated cluster **remote** 20 times. There is room for improvement in the case of many clusters. I will study some other confirmation modals to see if there is a pattern for better scrolling.